File: ruby/problem-2.rb
def even_fib_sum(n)
  fibs = [1, 2]
  while fibs[-1] < n
    fibs << fibs[-2] + fibs[-1]
  end
  fibs.select { |i| i < n && i.even? }.reduce(:+)
end

def even_fib_sum2(n)
  fibs = [1, 2]
  sum = 2
  while fibs[-1] < n
    next_fib = fibs[-1] + fibs[-2]
    fibs << next_fib
    sum += next_fib if next_fib.even? && next_fib < n
  end
  sum
end

p even_fib_sum(4000000)
p even_fib_sum2(4000000)

File: js/problem2.js
function evenFibSum(n) {
  var a = 2;
  var fibSum = 0;
  if (n > 1) {
    fibSum += 2;
  }
  var i = 3;
  while (i <= n) {
    if (i % 2 === 0) {
      fibSum += i;
    }
    var b = a
    a = i
    i += b
  }
  console.log(fibSum);
}
